mso/resource_mso_schema_site_vrf_route_leak.go

```diff
@@ -0,0 +1,252 @@
+package mso
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/container"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSOSchemaSiteVrfRouteLeak() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSOSchemaSiteVrfRouteLeakCreate,
+		Update: resourceMSOSchemaSiteVrfRouteLeakUpdate,
+		Read:   resourceMSOSchemaSiteVrfRouteLeakRead,
+		Delete: resourceMSOSchemaSiteVrfRouteLeakDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceMSOSchemaSiteVrfRouteLeakImport,
+		},
+
+		SchemaVersion: version,
+
+		Schema: (map[string]*schema.Schema{
+			"schema_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"template_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"site_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"vrf_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"target_vrf_schema_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"target_vrf_template_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"target_vrf_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"tenant_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "leak_all",
+				ValidateFunc: validation.StringInSlice([]string{
+					"leak_all",
+					"subnet_ip",
+					"all_subnet_ips",
+				}, false),
+			},
+			"subnet_ips": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		}),
+	}
+}
+
+func getVrfRef(schemaId, TemplateName, VrfName string) string {
+	return fmt.Sprintf("/schemas/%s/templates/%s/vrfs/%s", schemaId, TemplateName, VrfName)
+}
+
```

Comment:
If it's just a line, can it be inside getTargetVrfRef function?

Reply:
I separated it because I use this function also in the resourceMSOSchemaSiteVrfRouteLeakImport function. Arguably I think all these type of ref retrieval functions could be placed in utils or helper file.